pkg/lightning/restore/table_restore.go

```diff
@@ -694,15 +693,28 @@ func (tr *TableRestore) postProcess(
 					return false, err
 				}
 				if !needChecksum {
+					if rc.cfg.TikvImporter.DuplicateDetection {
+						if err := rc.backend.CollectLocalDuplicateRows(ctx, tr.encTable, rc.cfg.TiDB.SQLMode); err != nil {
+							tr.logger.Error("collect local duplicate keys failed", log.ShortError(err))
+						}
+					}
 					return false, nil
 				}
+				if rc.cfg.TikvImporter.DuplicateDetection {
+					if err := rc.backend.CollectRemoteDuplicateRows(ctx, tr.encTable, rc.cfg.TiDB.SQLMode); err != nil {
```

Comment:
It seems we need to avoid calling `CollectRemoteDuplicateRows` when backend is not `Local`. Or don't panic in `CollectRemoteDuplicateRows` when backend is `TiDB` and `Importer`.

Reply:
I will return an error when create other backend

Reply:
I will return an error when create other backend